### DISCUSSION TITLE
Removing Rich

### DISCUSSION THREAD

**Discussion by itsmeadarsh2008** [reflex-dev/reflex#3479] [Ideas]
Rich is too heavy with bloatware in the dependency. Would it be a better idea to make a progress bar using `tqdm` only because using rich doesn't make any sense with web apps in the production? It will be a good idea if we replace it with them and keep it simple. You can convert it to an issue if you want. Here are some example sentences. (No Boxes, Just Logs And These Sentences)
1.  ⚡ Compiling ETA: `{seconds}`
2. 🚧 Stopping the server
3. 🏓 Frontend is playing ping-pong on `{port}`
4. 🖥 Backend is serving food on `{port}` 
5. Your server is ready to go 🚀

**Comment by IamAdiSri** [comment]
Hi, I would like to work on this and was wondering about the extent to which this will change things. Are we completely removing `rich` from the codebase? Should the ETA be a dynamically changing number or should it just be printed once?

**Comment by ndb4263** [comment]
Hey, I would like to work on this bounty.

**Comment by ai-naymul** [comment]
@itsmeadarsh2008 , What's should I use in case of the logging? Should I use the python's print function or the tqdm's write function?

**Reply by itsmeadarsh2008** [reply]
Here's the issue https://github.com/reflex-dev/reflex/issues/3487. I guess you shouldn't use print. 
https://www.perplexity.ai/search/what-s-should-i-use-in-case-of-Pmt_VVy7SgO.8UyIgTGUxw#1

**Reply by ai-naymul** [reply]
> Here's the issue #3487. I guess you shouldn't use print. https://www.perplexity.ai/search/what-s-should-i-use-in-case-of-Pmt_VVy7SgO.8UyIgTGUxw#1

I think someone made a pr regarding this issue and there are some conflicts with the pr, should work on it individually?

**Reply by itsmeadarsh2008** [reply]
It just replaces the progress bar. Try to remove the rich completely. Work on your own I guess.

**Reply by ai-naymul** [reply]
> It just replaces the progress bar. Try to remove the rich completely. Work on your own I guess.

Please check out the following pr:
https://github.com/reflex-dev/reflex/pull/3773

**Reply by ai-naymul** [reply]
> It just replaces the progress bar. Try to remove the rich completely. Work on your own I guess.

Hi @itsmeadarsh2008 ,

I think there are no test cases neededd for this migration cause there were not also any test cases for the rich dependency and its just to show the progress bar, Am I right bro?

**Reply by itsmeadarsh2008** [reply]
Good evening. Test cases are needed whenever you change the internal structure of a project. Since, it's the CLI which shows the progress of compilation. You need to check whether all your test cases have passed or not. By the way, Try to search for rich all over the project make necessary migrations to another one (less-sized) like colorama, tqdm etc.
https://github.com/search?q=org%3Areflex-dev%20rich&type=code

**Reply by kraanzu** [reply]
Hey @itsmeadarsh2008, Is this issue still open? The author hasn't made any updates for a month :)

**Reply by itsmeadarsh2008** [reply]
@kraanzu yes, the issue is still open. you can contribute, or make a new PR instead.

**Reply by kraanzu** [reply]
Ok cool. Thanks for confirming @itsmeadarsh2008 ! :D

**Comment by dineshpabbati** [comment]
Hi,
I'd like to work on Removing Rich #3479.
